test/integration/ProjectBrowser.kt

```diff
@@ -0,0 +1,181 @@
+            wait(composeRule, 500)
+
+            composeRule.onNodeWithText("create_a_directory").assertExists()
+
+            StudioState.project.current!!.directory.entries.find { it.name == "create_a_directory" }!!.asDirectory()
+                .delete()
+            StudioState.project.current!!.reloadEntries()
+            composeRule.waitForIdle()
+            composeRule.onNodeWithText("create_a_directory").assertExists()
+        }
+    }
+
+    @Test
+    fun `Create a File`() {
+        val funcName = object{}.javaClass.enclosingMethod.name
+        runComposeRule(composeRule) {
+            setContent {
+                Studio.MainWindowContent(WindowContext(1000, 1000, 0, 0))
+            }
+            composeRule.waitForIdle()
+
+            cloneAndOpenProject(composeRule, SAMPLE_DATA_PATH, funcName)
+
+            StudioState.project.current!!.directory.asDirectory().tryCreateFile("file4")
+            wait(composeRule, 500)
+
+            StudioState.project.current!!.reloadEntries()
+            wait(composeRule, 500)
+
+
+            composeRule.onNodeWithText("file4").assertExists()
+
+            StudioState.project.current!!.directory.entries.find { it.name == "file4" }!!.asFile()
+                .delete()
+        }
+    }
+
+    @Test
+    fun `Rename a File`() {
+        val funcName = object{}.javaClass.enclosingMethod.name
+        runComposeRule(composeRule) {
+            setContent {
+                Studio.MainWindowContent(WindowContext(1000, 1000, 0, 0))
+            }
+            composeRule.waitForIdle()
+
+            cloneAndOpenProject(composeRule, SAMPLE_DATA_PATH, funcName)
+
+            StudioState.project.current!!.directory.entries.find { it.name == "file3" }!!.asFile().tryRename("file3_0")
+            wait(composeRule, 500)
+
+            StudioState.project.current!!.reloadEntries()
+            wait(composeRule, 500)
+
+            composeRule.onNodeWithText("file3_0").assertExists()
+            StudioState.project.current!!.directory.entries.find { it.name == "file3_0" }!!.asFile().tryRename("file3")
+            wait(composeRule, 500)
+
+        }
+    }
+
+    @Test
+    fun `Delete a File`() {
+        val funcName = object{}.javaClass.enclosingMethod.name
+        runComposeRule(composeRule) {
+            setContent {
+                Studio.MainWindowContent(WindowContext(1000, 1000, 0, 0))
+            }
+            composeRule.waitForIdle()
+
+            cloneAndOpenProject(composeRule, SAMPLE_DATA_PATH, funcName)
+
+            StudioState.project.current!!.directory.entries.find { it.name == "file3" }!!.asFile().delete()
+            wait(composeRule, 500)
+
+            StudioState.project.current!!.reloadEntries()
+            wait(composeRule, 500)
+
+            composeRule.onNodeWithText("file3").assertDoesNotExist()
+
+            StudioState.project.current!!.directory.asDirectory()
+                .tryCreateFile("file3")
+            delay(500)
```

Comment:
Using composeRule.waitForIdle isn't the same as StudioState.project.current!!.reloadEntries(), the latter has to be called manually to my knowledge. Sometimes delay() use is valid since we aren't waiting for recomposition - unless we decide it's acceptable to substitute in wait for delay for the sake of uniformity.

We could use wait here instead, but I'd like to discuss this further.

Reply:
It turns out we don't need this code anyway.